### ISSUE TITLE
Add Inno Setup to Visual Studio 2017 Images

### ISSUE THREAD

**Issue by nick-thompson** [Microsoft/azure-pipelines-image-generation#261]
A free, popular tool for creating installers for Windows applications.
http://www.jrsoftware.org/isinfo.php

**Comment by bryanmacfarlane** [comment]
@chrisrpatterson - I don't believe we're going to be adding (and supporting) software packages like this.

Your available options are to pull JIT (that agent runs as administrator).  You can even create a tool installer task and add it to the marketplace.

Run a private agent on an azure VM.

**Comment by nick-thompson** [comment]
Thanks @bryanmacfarlane. So I've been playing with this for the better part of the past week, I don't mind installing it myself on a hosted agent (it's pretty simple with a `curl` and executing the installer with `/silent`), but the agent hangs currently because it's waiting for permission.

If I can run the `vs2017-win2016` agent with administrator privileges so as to avoid that permission prompt then I imagine the installation will be plenty easy (on a local Win10 installation I've been able to curl & install and everything works as expected after permission is given). I don't quite understand what you mean by "pull JIT" though? This is my config, how would I modify it to pull JIT?

```
- job: Windows
  pool:
    vmImage: 'vs2017-win2016'
  steps:
  - task: MSBuild@1
    inputs:
      solution: 'projects/**/*.sln'
      platform: 'x64'
      configuration: 'release'
      clean: true
  - task: ShellScript@2
    inputs:
      scriptPath: ./scripts/azure-build_Windows.sh
```

(My `azure-build_Windows.sh` script is a bash script that curls and tries to install InnoSetup, that's where my build hangs currently)

And my understanding with the tool installers was that this was a preview feature that, currently, seems unavailable to me?

**Comment by bryanmacfarlane** [comment]
You would need to follow up with InnoSetup to find out why the installation hangs.  The agent on hosted runs with admin permissions.  JIT = Just in time which means install it as part of your build process which is what you're doing.

**Comment by nick-thompson** [comment]
Ah, right, thanks.

I've asked around in other forums about what's going on here, and I'm getting the suggestion that this is a "UAC" popup and not an authorization popup (or therefore not the installer). Apparently that's evident because the popup message is "Do you want to allow InnoSetup to make changes to your device?" (I'm primarily an OS X user, so this is all relatively unfamiliar to me). Is there a way to disable UAC on the build agents?

**Comment by nick-thompson** [comment]
Ok, I've taken my questions to the InnoSetup help forum and it's been quite helpful (http://news.jrsoftware.org/read/article.php?id=106698&group=jrsoftware.innosetup#106698).

@bryanmacfarlane can you tell me how the permissions operate on the Azure VM? When I execute a `ShellScript@2` task on the `vs2017-win2016` pool, am I running in an elevated prompt? Or `CmdLine@2` on the same pool? If not, is there a way (such as `runas`) to kick off executing the installer in a new, elevated prompt?

**Comment by bryanmacfarlane** [comment]
@ericsciple for above question

**Comment by ericsciple** [comment]
on the vs2017 pool I believe everything should be running elevated

**Comment by brcrista** [comment]
Resolved with #313 

@nick-thompson for future reference, the VS2017 image has Chocolatey installed on it.  That's usually a good way to install tools during a build if you can find your tool as a package there.
